lib/src/helpers/lists.dart

```diff
@@ -5,40 +5,41 @@
 import 'dart:collection';
 import 'dart:js_interop';
 
-/// Defines interop members on a JSObject that would be present on a js list object.
-/// The JSObject is assumed to be one of the js list type as right now
-/// we don't have a single interface that would represent all the js list types.
-extension on JSObject {
+/// Defines interop members on a JSObject that would be present on a JS list object.
+/// The JSObject is assumed to be one of the JS list type as right now
+/// we don't have a single interface that would represent all the JS list types.
+extension type _JSList<T extends JSObject>(JSObject _) implements JSObject {
   /// The **`item()`** method returns the [JSObject]
   /// at the specified index in the list.
-  external JSObject item(int index);
+  external T item(int index);
 
   /// The **`length`** read-only property indicates the number of
   /// items in a given list.
   external int get length;
 }
 
+/// A wrapper to present a JS immutable list of type T as a `List<U>` where U
```

Comment:
nit: \`T\` and \`U\`

Reply:
How's this for updated definition? 

```A wrapper to present a JS immutable list of type T and list item type U as a `List<U>`.```

Reply:
Sure, that works. My comment above was mostly about adding backticks to `T` and `U` so they'd be treated as code.